lib/mine.js

```diff
@@ -4,26 +4,22 @@
  */
 
 // const schedule = require('node-schedule')
-global.config = require('./config')
-const Sonar = require('./lib/sonar')
+global.config = require('../config')
+const Sonar = require('./sonar')
 const ipfsAPI = require('ipfs-api')
 const tmp = require('tmp')
 const path = require('path')
 const fs = require('fs')
 const spawn = require('child_process').spawn
 const Web3 = require('web3')
 
-const web3 = new Web3(new Web3.providers.HttpProvider(Web3.givenProvider || 'http://localhost:8545'))
+async function checkForModels (mineAddress, contractAddress, ethereumUrl, ipfsUrl) {
+  const web3 = new Web3(new Web3.providers.HttpProvider(ethereumUrl || Web3.givenProvider || 'http://localhost:8545'))
```

Comment:
What's the significance of `givenProvider` atm? Eth docs make it seem like thats not pertinent.